Frameworks/UIKit/UIGestureRecognizer.mm

```diff
@@ -231,24 +229,19 @@ - (void)requireGestureRecognizerToFail:(UIGestureRecognizer*)gesture {
     UNIMPLEMENTED();
 }
 
-+ (BOOL)_fireGestures:(id)gestures shouldCancelTouches:(BOOL&)shouldCancelTouches {
-    bool didRecognize = false;
-
-    for (UIGestureRecognizer* curgesture in gestures) {
-        UIGestureRecognizerState state = (UIGestureRecognizerState)[curgesture state];
++ (BOOL)_fireGesture:(UIGestureRecognizer*)gesture {
+    UIGestureRecognizerState state = [gesture state];
 
-        if (state == UIGestureRecognizerStateRecognized || state == UIGestureRecognizerStateBegan ||
-            state == UIGestureRecognizerStateChanged || state == UIGestureRecognizerStateEnded) {
-            [curgesture _fire];
-            shouldCancelTouches |= curgesture.cancelsTouchesInView;
-            didRecognize = true;
-        }
+    if (state == UIGestureRecognizerStateRecognized || state == UIGestureRecognizerStateBegan || state == UIGestureRecognizerStateChanged ||
+        state == UIGestureRecognizerStateEnded) {
+        [gesture _fire];
+        return YES;
     }
 
-    return didRecognize;
+    return NO;
 }
 
-- (void)cancel {
+- (void)_cancel {
     if (_state == UIGestureRecognizerStateBegan || _state == UIGestureRecognizerStateChanged) {
         _state = UIGestureRecognizerStateCancelled;
```

Comment:
>_state = UIGestureRecognizerStateCancelled; [](start = 8, length = 43)

Since this state is set in both conditions, you could move it outside the if-else block and set it always.